test/puppetlabs/puppetdb/http/index_test.clj

```diff
@@ -179,13 +190,13 @@
           (is (= results #{{:certname "host2"}})))))
 
     (testing "nodes"
-      (testing "query should return all nodes (including deactivated ones)"
+      (testing "query should return only active nodes"
         (doseq [query [["from" "nodes"]
                        "nodes {}"]]
           (is (= (set (mapv :certname (query-result method endpoint query {})))
                  #{"host1" "host2" "host3"}))))
 
-      (testing "broad regexp query should return all nodes"
+      (testing "broad regexp query should return all active nodes"
```

Comment:
+1 to having a toggle on this behavior. Some more info in PDB-2642.

In most cases, folks only want results from active nodes. However, there are usecases where folks submit a query like `SELECT * FROM nodes` and expect the results  of _that exact query_. Not `SELECT * FROM nodes WHERE expired IS NOT NULL AND deactivated IS NOT NULL`, because that's not the query they submitted. For the API endpoints that do apply this sort of implicit filtering, folks who want a complete query result have to submit two separate queries: one where `active` is set to true and one where it is set to false. Then they have to figure out how to JOIN the two datasets outside of the Puppet tooling (custom scripting).

A concrete usecase for this is folks that have nodes in their population where Puppet doesn't run on a schedule. Those nodes will perpetually be "inactive" because puppet is only executed during specific, infrequent, change windows. Running reports that include data from every node in this sort of population requires a query where the implicit filter is not used.

This implicit filter is a good thing and probably what the majority of folks want for the majority of usecases. However, there needs to be a way to conditionally disable this behavior (perhaps via a query parameter in the URL). Otherwise, this change will create a regression for the folks who need complete results from the entire population in a single query.